Mvp 52/알림메시지 기능구현

hojkim77 opened the pull request:
TODO
1. "알림"관련 로직 작성
알림스크린(notification-screen) ui 및 로직 작성.
이에 따른 api, MST 작성.

2. 푸시알림 저장
expo-notification, expo-device를 사용하여 각 디바이스의 푸시토큰 받아오는 함수 "util 폴더"에 작성.
위 함수를 사용하여 받아온 푸시토큰 값을 
    1. MST 저장
    2. POST ../user/push-token에 넘김

hojkim77 committed:
MVP-52 스크린 내 알림 컴포넌트 생성

hojkim77 committed:
MVP-52 알림컴포넌트 디자인

hojkim77 committed:
MVP-52 (진행중)알림컴포넌트 디자인(수정)

hojkim77 committed:
MVP-52 (진행중)notification API작성(백엔드측 핫픽스 후 수정 필요)

hojkim77 committed:
MVP-52 (진행중)api 완성 및 model 작성

hojkim77 committed:
MVP-52 (진행중)수정 필요 변경사항

hojkim77 committed:
MVP-52 (진행중)알림 확인 로직 완성

hojkim77 committed:
MVP-52 (진행중)header디자인 및 삭제하기버튼로직 생성

hojkim77 committed:
MVP-52 (진행중)알림 삭제하기 model 완성

hojkim77 committed:
MVP-52 푸시알림 저장(api, MST 모두)

smnchoi commented on app/utils/get-pushToken.ts:
```diff
@@ -0,0 +1,35 @@
+      finalStatus = status
+    }
+    if (finalStatus !== "granted") {
+      alert("Failed to get push token for push notification!")
```
@hojkim77 
- 푸시 토큰을 얻지 못했을 경우 알림 메시지를 한글로 교체해주세요. 
- alert 대신 alertModal 을 사용해주세요

smnchoi commented on app/utils/get-pushToken.ts:
```diff
@@ -0,0 +1,35 @@
+      name: "default",
+      importance: Notifications.AndroidImportance.MAX,
+      vibrationPattern: [0, 250, 250, 250],
+      lightColor: "#FF231F7C",
```
@hojkim77 
- `lightColor: GIVER_CASUAL_NAVY` 로 변경해주세요 :)

smnchoi commented on app/models/user-store/user-store.ts:
```diff
@@ -158,7 +159,13 @@ export const UserStoreModel = types
         }
-
+        // 푸시토큰 발급
+        const pushToken = await registerForPushNotificationsAsync().then((token) => {
```
@hojkim77 
# 항상 "예외처리" 를 유념해주세요
- registerForPushNotificationsAsync() 함수가 실패할 경우에는 어떻게 되나요?
 - 그 경우, token 은 null 인가요? "" 엠티스트링인가요?
 - 이때에도(실패한 경우에도) postPushToken 을 호출하는 것이 의도대로 작동하는 것인가요?
 - 만약, 실패시에도 postPushToken 을 호출하는 것이 호중님이 의도한 바가 아니라면, 조건문을 통해 예외를 처리해주세요.

hojkim77 committed:
MVP-52 PR피드백 수정

smnchoi commented:
@hojkim77 완벽하네요. 업무 진행해주시느라 고생많으셨습니다 :) 
내일 아침중으로 컨플릭트 해결해서 머지하도록 하겠습니다. 수고많으셨습니다! 
호중님 시험기간 일정 고려하여 새 업무 드리도록 하겠습니다 👍

smnchoi committed:
MVP-52 Merge Conflicts 해결 - Merge branch 'develop' into MVP-52/알림메시지-기능구현'

# Conflicts:
#	app/models/index.ts
#	app/models/root-store/root-store.ts
#	app/navigators/cl-stack-navigator.tsx
#	app/screens/index.ts
#	app/screens/search-stack/search-screen/search-screen.tsx

smnchoi committed:
MVP-52 Fix: import 경로 에러 수정, registerForPushNotificationsAsync() 시뮬레이터일때 return undefined 수정, notificationStore 초기화 버튼 notification-screen 으로 이전 함
